### ISSUE TITLE
CreateBuiltinFunction’s signature has changed

### ISSUE THREAD

**Issue by bathos** [whatwg/webidl#1106]
ECMA-262 [changed the signature](https://github.com/tc39/ecma262/pull/2116) of the [CreateBuiltinFunction](https://tc39.es/ecma262/#sec-createbuiltinfunction) last year. Perhaps there were changes on both sides around the same time or something, though, since the PR says it “preserves compatibility for WebIDL,” but it seems all 30 usages in Web IDL are incorrect now.

```
PREVIOUS SIGNATURE: ( steps, internalSlotsList [ , realm [ , prototype ] ] )
CURRENT SIGNATURE:  ( behaviour, length, name, additionalInternalSlotsList [ , realm [ , prototype [ , prefix ] ] ] )
```

Web IDL usages all look like one of these:

```
CreateBuiltinFunction(steps, « [[Unforgeables]] », realm, constructorProto).
CreateBuiltinFunction(steps, « », realm)
CreateBuiltinFunction(steps, « »)
```

Explicit names and lengths are required. In cases where Web IDL is already setting them afterwards, it’d just become a single call:

```
// ...where it’s currently like this:
Let F be ! CreateBuiltinFunction(steps, « », realm).
Perform ! SetFunctionName(F, id).
Perform ! SetFunctionLength(F, 0).

// ...it should be like this:
Let F be ! CreateBuiltinFunction(steps, 0, id, « », realm).
```

Web IDL doesn’t always set a name or length, though. I’m not certain what’s right for those cases, but it looks like they’re all Promise-related and mirror patterns in ECMA-262. Since the ES versions already use the new signature, it’s likely their values (e.g. empty string for name) are what’s wanted here, too.

- [Let onFulfilled be ! {...web idl ...}](https://webidl.spec.whatwg.org/#dfn-perform-steps-once-promise-is-settled) (at step 4)
- [Let onFulfilled be ! {...es...}](https://tc39.es/ecma262/#sec-asyncfromsynciteratorcontinuation) (at step 8)

**Comment by bathos** [comment]
Oh, I’d only searched issues earlier and missed that there’s [an open PR](https://github.com/whatwg/webidl/pull/914) meant to address this (which is from the same time, explaining the comment). I suppose it makes sense to leave the issue open though as it’s still pending.

**Comment by domenic** [comment]
As long as you are willing to enter the name you sign legal contracts with, into a form on the internet, you are able to fill out the participant agreement.

**Comment by bathos** [comment]
Would that name would appear in the public https://github.com/whatwg/participant-data/blob/main/individuals.json? It seemed like you had a good solution in that thread (a distinct participant-data-private), but it wasn’t clear to me that it had been adopted.

**Comment by domenic** [comment]
Yes, it would appear publicly. If that's not acceptable then indeed you too are stuck :(
